cfme/cloud/provider/azure.py

```diff
@@ -1,47 +1,59 @@
+from mgmtsystem.azure import AzureSystem
+
 from . import CloudProvider
+from cfme.common.provider import DefaultEndpoint, DefaultEndpointForm
 from utils.version import pick
-from mgmtsystem.azure import AzureSystem
+
+
+class AzureEndpoint(DefaultEndpoint):
+    @property
+    def view_value_mapping(self):
+        return {}
+
+
+class AzureEndpointForm(DefaultEndpointForm):
+    pass
 
 
 class AzureProvider(CloudProvider):
     type_name = "azure"
     mgmt_class = AzureSystem
     db_types = ["Azure::CloudManager"]
+    endpoints_form = AzureEndpointForm
 
-    def __init__(self, name=None, credentials=None, zone=None, key=None, region=None,
+    def __init__(self, name=None, endpoints=None, zone=None, key=None, region=None,
                  tenant_id=None, subscription_id=None, appliance=None):
-        super(AzureProvider, self).__init__(name=name, credentials=credentials,
+        super(AzureProvider, self).__init__(name=name, endpoints=endpoints,
                                             zone=zone, key=key, appliance=appliance)
         self.region = region  # Region can be a string or a dict for version pick
         self.tenant_id = tenant_id
         self.subscription_id = subscription_id
 
-    def _form_mapping(self, create=None, **kwargs):
-        region = kwargs.get('region')
-        if isinstance(region, dict):
-            region = pick(region)
-        # Will still need to figure out where to put the tenant id.
-        return {'name_text': kwargs.get('name'),
-                'type_select': create and 'Azure',
-                'region_select': region,
-                'azure_tenant_id': kwargs.get('tenant_id'),
-                'azure_subscription_id': kwargs.get('subscription_id')}
+    @property
+    def view_value_mapping(self):
+        region = pick(self.region) if isinstance(self.region, dict) else self.region
```

Comment:
I'm very anti. PEP8 has nothing against conditional expressions and those are handy when statement isn't complex.